### PR TITLE
Improve docstrings for std and var

### PR DESCRIPTION
**Pull request by nalimilan**
```[ci skip]```

### CODE REVIEW

File: stdlib/Statistics/src/Statistics.jl [hunk 1]
```diff
@@ -276,15 +281,17 @@ end
 
 
 """
-    var(v; dims, corrected::Bool=true, mean=nothing)
+    var(itr; dims, corrected::Bool=true, mean=nothing)
 
-Compute the sample variance of a vector or array `v`, optionally along the given dimensions.
```
Comment (nalimilan) [inline stdlib/Statistics/src/Statistics.jl]: This part was actually incorrect (out of date).

File: stdlib/Statistics/src/Statistics.jl [hunk 2]
```diff
@@ -248,13 +248,18 @@ function varm!(R::AbstractArray{S}, A::AbstractArray, m::AbstractArray; correcte
...
 """
-    varm(v, m; dims, corrected::Bool=true)
+    varm(itr, m; dims, corrected::Bool=true)
 
-Compute the sample variance of a collection `v` with known mean(s) `m`,
-optionally over the given dimensions. `m` may contain means for each dimension of
-`v`. If `corrected` is `true`, then the sum is scaled with `n-1`,
+Compute the sample variance of collection `itr`, with known mean(s) `m`.
+The algorithm will return an estimator of the generative distribution's variance
+under the assumption that each entry of `v` is an IID drawn from that generative
```
Comment (StefanKarpinski) [inline stdlib/Statistics/src/Statistics.jl]:
```suggestion
under the assumption that each entry of `itr` is an IID drawn from that generative
```

File: stdlib/Statistics/src/Statistics.jl [hunk 3]
```diff
@@ -248,13 +248,18 @@ function varm!(R::AbstractArray{S}, A::AbstractArray, m::AbstractArray; correcte
...
+distribution. This computation is equivalent to calculating `sum(abs2, v - mean(v)) /
```
Comment (StefanKarpinski) [inline stdlib/Statistics/src/Statistics.jl]:
```suggestion
distribution. This computation is equivalent to calculating `sum(abs2, itr .- mean(itr)) /
```

File: stdlib/Statistics/src/Statistics.jl [hunk 4]
```diff
@@ -248,13 +248,18 @@ function varm!(R::AbstractArray{S}, A::AbstractArray, m::AbstractArray; correcte
...
+(length(v) - 1)`. If `corrected` is `true`, then the sum is scaled with `n-1`,
```
Comment (StefanKarpinski) [inline stdlib/Statistics/src/Statistics.jl]:
```suggestion
(length(itr) - 1)`. If `corrected` is `true`, then the sum is scaled with `n-1`,
```

File: stdlib/Statistics/src/Statistics.jl [hunk 5]
```diff
@@ -248,13 +248,18 @@ function varm!(R::AbstractArray{S}, A::AbstractArray, m::AbstractArray; correcte
...
 whereas the sum is scaled with `n` if `corrected` is `false` where `n = length(v)`.
```
Comment (StefanKarpinski) [inline stdlib/Statistics/src/Statistics.jl]:
```suggestion
whereas the sum is scaled with `n` if `corrected` is `false` where `n = length(itr)`.
```

File: stdlib/Statistics/src/Statistics.jl [hunk 6]
```diff
@@ -276,15 +281,17 @@ end
 
 
 """
-    var(v; dims, corrected::Bool=true, mean=nothing)
+    var(itr; dims, corrected::Bool=true, mean=nothing)
 
-Compute the sample variance of a vector or array `v`, optionally along the given dimensions.
+Compute the sample variance of collection `itr`.
 The algorithm will return an estimator of the generative distribution's variance
 under the assumption that each entry of `v` is an IID drawn from that generative
```
Comment (StefanKarpinski) [inline stdlib/Statistics/src/Statistics.jl]:
```suggestion
under the assumption that each entry of `itr` is an IID drawn from that generative
```

File: stdlib/Statistics/src/Statistics.jl [hunk 7]
```diff
@@ -276,15 +281,17 @@ end
...
 distribution. This computation is equivalent to calculating `sum(abs2, v - mean(v)) /
```
Comment (StefanKarpinski) [inline stdlib/Statistics/src/Statistics.jl]:
```suggestion
distribution. This computation is equivalent to calculating `sum(abs2, itr - mean(itr)) /
```

File: stdlib/Statistics/src/Statistics.jl [hunk 8]
```diff
@@ -276,15 +281,17 @@ end
...
 (length(v) - 1)`. If `corrected` is `true`, then the sum is scaled with `n-1`,
```
Comment (StefanKarpinski) [inline stdlib/Statistics/src/Statistics.jl]:
```suggestion
(length(itr) - 1)`. If `corrected` is `true`, then the sum is scaled with `n-1`,
```

File: stdlib/Statistics/src/Statistics.jl [hunk 9]
```diff
@@ -276,15 +281,17 @@ end
...
 whereas the sum is scaled with `n` if `corrected` is `false` where `n = length(v)`.
```
Comment (StefanKarpinski) [inline stdlib/Statistics/src/Statistics.jl]:
```suggestion
whereas the sum is scaled with `n` if `corrected` is `false` where `n = length(itr)`.
```

**Review by StefanKarpinski — CHANGES_REQUESTED** [review]
You need to change a lot more instances of `v` to `itr` for this to be a consistent change. I suggested a few of them but there were more so I stopped after a while. Note that the "equivalent code" looks a bit weird since `length(itr)` is not generally a thing you can do for an iterable.

**Comment by nalimilan** [comment]
Woops, sorry, that's the kind of change that sounds trivial but ends up requiring more though. I've added "For arrays, this is equivalent to" and avoided using `length` where it can be described using words.

**Review by StefanKarpinski — APPROVED** [review]
Looks good now.

**Pull request merged**